test/Ocelot.AcceptanceTests/PollyQoSTests.cs

```diff
@@ -1,276 +1,311 @@
+
+        this.Given(x => x.GivenThereIsABrokenServiceRunningOn($"http://localhost:{port}"))
+            .And(x => _steps.GivenThereIsAConfiguration(configuration))
+            .And(x => _steps.GivenOcelotIsRunningWithPolly())
+            .And(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
+            .And(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
+            .And(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
+            .Then(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.ServiceUnavailable))
+            .BDDfy();
+    }
+
+    [Fact]
+    public void Should_open_circuit_breaker_then_close()
+    {
+        var port = PortFinder.GetRandomPort();
+
+        var configuration = new FileConfiguration
+        {
+            Routes = new List<FileRoute>
+            {
+                new()
+                {
+                    DownstreamPathTemplate = "/",
+                    DownstreamScheme = "http",
+                    DownstreamHostAndPorts = new List<FileHostAndPort>
+                    {
+                        new()
+                        {
+                            Host = "localhost",
+                            Port = port,
+                        },
+                    },
+                    UpstreamPathTemplate = "/",
+                    UpstreamHttpMethod = new List<string> { "Get" },
+                    QoSOptions = new FileQoSOptions
+                    {
+                        ExceptionsAllowedBeforeBreaking = 1,
+                        TimeoutValue = 500,
+                        DurationOfBreak = 1000,
+                    },
+                },
+            },
+        };
+
+        this.Given(x => x.GivenThereIsAPossiblyBrokenServiceRunningOn($"http://localhost:{port}", "Hello from Laura"))
+            .Given(x => _steps.GivenThereIsAConfiguration(configuration))
+            .Given(x => _steps.GivenOcelotIsRunningWithPolly())
+            .When(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
+            .Then(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.OK))
+            .And(x => _steps.ThenTheResponseBodyShouldBe("Hello from Laura"))
+            .Given(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
+            .Given(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.ServiceUnavailable))
+            .Given(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
+            .Given(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.ServiceUnavailable))
+            .Given(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
+            .Given(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.ServiceUnavailable))
+            .Given(x => GivenIWaitMilliseconds(3000))
+            .When(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
+            .Then(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.OK))
+            .And(x => _steps.ThenTheResponseBodyShouldBe("Hello from Laura"))
+            .BDDfy();
+    }
+
+    [Fact]
+    public void Open_circuit_should_not_effect_different_route()
```

Comment:
Did you fix this #1706 unstable test? 😍 
Last fail which known for me is build [2454](https://circleci.com/gh/ThreeMammals/Ocelot/2454)

Reply:
I hope so, I simplified the logic yes.

Reply:
Thanks!